### PR TITLE
Documentation for using the widget

### PR DESCRIPTION
**Pull request by Nking92**
Rough draft of the documentation for using the widget. For issue https://github.com/microsoft/BackwardCompatibilityML/issues/15

### CODE REVIEW

File: docs/backwardcompatibilityml/topics/using_the_compatibility_analysis_widget.rst [hunk 1]
```diff
@@ -3,8 +3,129 @@
...
+How to Use the CompatibilityAnalysis API
+----------------------------------------
+
+The CompatibilityAnalysis API has a large number of parameters. To view the Python documentation
+for the API, execute ``?CompatibilityAnalysis`` within a Jupyter notebook. The documentation  for all of the parameters will be displayed at the bottom of the window.
+
+In this article, we will reference the ``compatibility-analysis-cifar10-resnet18`` example notebook.
+You can find this notebook at ``./examples/compatibility-analysis-cifar10-resnet18`` from the
+BackwardCompatibilityML project root. This notebook uses the widget to compare the performance of 
+resnet18 with the following 6 layer network using the cifar10 dataset::
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_compatibility_analysis_widget.rst:21]: Change "6 layer" to "six-layer."

Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_compatibility_analysis_widget.rst:21]: Change "6 layer" to "six-layer."

File: docs/backwardcompatibilityml/topics/using_the_compatibility_analysis_widget.rst [hunk 2]
```diff
@@ -3,8 +3,129 @@
...
+
+The two scatter plots graph the backward compatibility of the model against the model accuracy for a particular
+value of |lambda|:sub:`c`. Hovering over a point shows the value of |lambda|:sub:`c` for that point. Clicking on a point
+loads detailed results and error analysis for that particular value of |lambda|:sub:`c`.
+
+The numeric values for BTC, BEC, model accuracy, and |lambda|:sub:`c` are shown in a table in the middle of the widget.
+Below that table, there is a Venn diagram and a histogram that plot the errors made by each model.
+The Venn diagram shows the intersection of errors made by the previous model with errors made by the new model.
+The red region represents errors made only by the new model, the yellow region represents errors made by both models,
+and the green region represents error made only by the old model. The histogram breaks down incompatible data points by class.
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_compatibility_analysis_widget.rst:88]: Add an "s" to "error."

File: docs/backwardcompatibilityml/topics/using_the_compatibility_analysis_widget.rst [hunk 3]
```diff
@@ -3,8 +3,129 @@
...
+
+In the example below, class 0 has been selected in the histogram. The mislabeled pictures are shown in the
+table underneath. Notice that h1's predictions match the ground truth for each data point while h2's predictions
+do not. This is what we would expect to see based on our definition of incompatible points.
+
+.. image:: ../media/error_instances_table.png
+    :width: 700
+
+The CompatibilityAnalysis API contains two optional parameters, ``get_instance_metadata`` 
+and ``get_instance_image_by_id``, that make the data shown in the table more descriptive.
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_compatibility_analysis_widget.rst:104]: Change "that" to "which."

**Review by KathleenW1 — COMMENTED** [review]
Several minor edits indicated in in-line comments.

File: docs/backwardcompatibilityml/topics/using_the_compatibility_analysis_widget.rst [hunk 4]
```diff
@@ -3,8 +3,129 @@
 Using the Backward Compatibility ML Compatibility Analysis Widget
 =================================================================
 
-.. note::
-    This topic will be added in a future release. Will cover how to
-    use the Compatibility Analysis widget and visualizations to 
-    interpret outputs from loss functions
+The compatibility analysis widget can be used to quickly determine which loss
+function and value of |lambda|:sub:`c` performs best for your models. The widget will
+make a sweep of the possible values for |lambda|:sub:`c` and show a visualization of the
```
Comment (jlema) [inline docs/backwardcompatibilityml/topics/using_the_compatibility_analysis_widget.rst:8]: Please mention that it also sweeps over the two selected loss functions.

**Review by jlema — CHANGES_REQUESTED** [review]

**Review by jlema — APPROVED** [review]

**Review by KathleenW1 — APPROVED** [review]

**Review by nushib — APPROVED** [review]

**Pull request merged**